### DISCUSSION TITLE
Preferences storage

### DISCUSSION THREAD

**Discussion by wwitzel3** [vmware-tanzu/octant#2694] [General]
How do we want to handle preference storage in Octant?
- PreferencesStateManager on the Octant side loaded with content at startup?
- Direct read from a file?

Other thoughts?

**Comment by wwitzel3** [comment]
What if we had a TOML file?

```
[system]
context = "my-local-cluster"
namespace = "ns-1"
namespace_list = [ "ns-1", "ns-2", "ns-3"]
disable_cluster_overview = true
kubeconfig = "/path/to/kubeconfig"
plugin_path = "/path/to/plugin/folder"

[preferences]
theme = "dark"
navigation_expanded = true
navigatiion_text = false
finalize_thersholed = "10m"
```

Potentially even expand it to explicit resource listings for heavily restricted clusters.

```
[[resources]]
version = "v1"
kind = "Pod"

[[resources]]
group = "apps"
version = "v1"
kind = "Deployment"
```

**Comment by wwitzel3** [comment]
My thought is Octant would have a file watcher, making the octant server process being purely a consumer of the TOML file and the electron frontend would handle all of the setting / updating.

**Reply by karayim** [reply]
I support the idea of a TOML file as long as nothing stored there is a privacy concern.
